src/wallet/api/wallet.cpp

```diff
@@ -2058,544 +1986,611 @@ bool WalletImpl::verifySignedMessage(const std::string &message, const std::stri
     }
-    bool ret = m_wallet->unblackball_output(std::make_pair(raw_amount, raw_offset));
-    if (!ret)
-    {
-        setStatusError(tr("Failed to mark output as unspent"));
-        return false;
+
+    return 0;
+}
+
+bool WalletImpl::hasMultisigPartialKeyImages() const {
+    try {
+        clearStatus();
+        checkMultisigWalletReady(m_wallet);
+
+        return m_wallet->has_multisig_partial_key_images();
+    } catch (const exception& e) {
+        LOG_ERROR("Error on checking for partial multisig key images: " << e.what());
+        setStatusError(string(tr("Failed to check for partial multisig key images: ")) + e.what());
     }
-    return true;
+
+    return false;
 }
 
-bool WalletImpl::getRing(const std::string &key_image, std::vector<uint64_t> &ring) const
-{
-    crypto::key_image raw_key_image;
-    if (!epee::string_tools::hex_to_pod(key_image, raw_key_image))
-    {
-        setStatusError(tr("Failed to parse key image"));
-        return false;
-    }
-    bool ret = m_wallet->get_ring(raw_key_image, ring);
-    if (!ret)
-    {
-        setStatusError(tr("Failed to get ring"));
-        return false;
+PendingTransaction* WalletImpl::restoreMultisigTransaction(const string& signData) {
+    try {
+        clearStatus();
+        checkMultisigWalletReady(m_wallet);
+
+        string binary;
+        if (!epee::string_tools::parse_hexstr_to_binbuff(signData, binary)) {
+            throw runtime_error("Failed to deserialize multisig transaction");
+        }
+
+        tools::wallet2::multisig_tx_set txSet;
+        if (!m_wallet->load_multisig_tx(binary, txSet, {})) {
+          throw runtime_error("couldn't parse multisig transaction data");
+        }
+
+        auto ptx = new PendingTransactionImpl(*this);
+        ptx->m_pending_tx = txSet.m_ptx;
+        ptx->m_signers = txSet.m_signers;
+
+        return ptx;
+    } catch (exception& e) {
+        LOG_ERROR("Error on restoring multisig transaction: " << e.what());
+        setStatusError(string(tr("Failed to restore multisig transaction: ")) + e.what());
     }
-    return true;
+
+    return nullptr;
 }
 
-bool WalletImpl::getRings(const std::string &txid, std::vector<std::pair<std::string, std::vector<uint64_t>>> &rings) const
+std::string WalletImpl::signMultisigParticipant(const std::string &message) const
 {
-    crypto::hash raw_txid;
-    if (!epee::string_tools::hex_to_pod(txid, raw_txid))
-    {
-        setStatusError(tr("Failed to parse txid"));
-        return false;
-    }
-    std::vector<std::pair<crypto::key_image, std::vector<uint64_t>>> raw_rings;
-    bool ret = m_wallet->get_rings(raw_txid, raw_rings);
-    if (!ret)
-    {
-        setStatusError(tr("Failed to get rings"));
-        return false;
+    clearStatus();
+
+    bool ready = false;
+    if (!m_wallet->multisig(&ready) || !ready) {
+        m_status = Status_Error;
+        m_errorString = tr("The wallet must be in multisig ready state");
```

Comment:
There is a semantical change here. The original form is here [1].  This new style condition is not deterministically the same as the original form.

1. https://github.com/monero-project/monero/pull/9368/files#diff-fcae134993cca2b6f1e887a60638f226ab00165e30bd4d326c17f0a58e11871fL2068

Reply:
Thanks for taking a look and good catch. This is not intentional and probably snuck in when rebased. Change was pretty recent https://github.com/monero-project/monero/commit/3a47cdad13fd8486b7d5559ec004ff49b3b3aa6d

Regarding your comment below, this PR is already supposed to be the broken down version of a bigger/messier PR, so if you find anything that's not "ordering functions", it does not belong, but creeped in. Except removing some comments which would get removed with the next PR, because I thought having two PRs (first move comment, second remove comment) will also be unnecessary reviewers work.